From aa22632d8cef3c04cfcabd83b21b9adbc06e780e Mon Sep 17 00:00:00 2001
From: Igor Sugak <sugak@fb.com>
Date: Sat, 7 Jan 2017 13:14:41 -0800
Subject: [PATCH] fix build with boost 1.63.0

Summary:
When building with boost 1.63.0 exposed:
```
proxygen/lib/http/codec/HTTP2Codec.cpp:1056:11: error: no viable overloaded '='
      pri = {std::get<0>(*res), std::get<1>(*res), std::get<2>(*res)};
      ~~~ ^ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

Reviewed By: yfeldblum

Differential Revision: D4390928

fbshipit-source-id: 1ffe4526946fa3e3be4155fd09141561c8cf57b0
---
 proxygen/lib/http/codec/HTTP2Codec.cpp | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/proxygen/lib/http/codec/HTTP2Codec.cpp b/proxygen/lib/http/codec/HTTP2Codec.cpp
index 7a7a914..dff1cc7 100644
--- a/proxygen/lib/http/codec/HTTP2Codec.cpp
+++ b/proxygen/lib/http/codec/HTTP2Codec.cpp
@@ -1053,7 +1053,8 @@ void HTTP2Codec::generateHeader(folly::IOBufQueue& writeBuf,
     auto res = msg.getHTTP2Priority();
     size_t split = kHeaderSplitSize;
     if (res) {
-      pri = {std::get<0>(*res), std::get<1>(*res), std::get<2>(*res)};
+      pri = http2::PriorityUpdate{std::get<0>(*res), std::get<1>(*res),
+                                  std::get<2>(*res)};
       if (split > http2::kFramePrioritySize) {
         split -= http2::kFramePrioritySize;
       }
